modules/apps/dynamic-data-lists/dynamic-data-lists-service/src/main/java/com/liferay/dynamic/data/lists/internal/change/tracking/spi/reference/DDLRecordTableReferenceDefinition.java

```diff
@@ -68,6 +69,9 @@ public void defineChildTableReferences(
 						DDMFieldTable.INSTANCE.storageId)
 				)
 			)
+		).singleColumnReference(
+			DDLRecordTable.INSTANCE.companyId,
+			DDMStorageLinkTable.INSTANCE.companyId
```

Comment:
We need to scope it down to get just the DDMStorageLinks that are associated with the DDLRecord. This query returns all DDMStorageLinks in the company.